src/error.ts

```diff
@@ -108,6 +108,16 @@ export interface ErrorDescription extends Document {
   errInfo?: Document;
 }
 
+function isAggregateError(e: unknown): e is Error & { errors: Error[] } {
+  return (
+    typeof e === 'object' &&
+    e != null &&
+    e instanceof Error &&
+    'errors' in e &&
+    Array.isArray(e.errors)
+  );
+}
```

Comment:
Do we need to confirm `instanceof Error`?

I think we should change the check here a bit, instead of focusing on "AggregateError" specifically, try and confirm more of the contract we _need_ from the object we're given

- [x] It's a non-null object
- [ ] If it has a `message` property, then it is an empty string
  - AggregateErrors can have a message, and if they do have one it should be preferred over building the joined string.
- [x] It has an `errors` property
- [ ] `errors` is a non-zero array
  - The non-zero check is to make sure that our `errors.map` actually returns something, but also we have no information to go off of, so maybe if we reach this case we can put some non-empty default, "see cause" or "good luck".